src/main/java/de/letsdev/maven/plugins/ios/mojo/BaseMojo.java

```diff
@@ -419,33 +474,124 @@
         this.addProperty(properties, Utils.PLUGIN_PROPERTIES.XCTEST_SDK.toString(), this.xcTestsSdk);
-        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.XCTEST_BUILD_ARGUMENTS.toString(), this.xcTestsBuildArguments);
-        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.RESET_SIMULATORS.toString(), Boolean.toString(this.resetSimulators));
+        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.XCTEST_BUILD_ARGUMENTS.toString(),
+                this.xcTestsBuildArguments);
+        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.RESET_SIMULATORS.toString(),
+                Boolean.toString(this.resetSimulators));
+        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.DERIVED_DATA_PATH.toString(), this.derivedDataPath);
+        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.XCTEST_DERIVED_DATA_PATH.toString(),
+                this.xcTestsDerivedDataPath);
+        this.addProperty(properties, Utils.PLUGIN_PROPERTIES.PROVISIONING_PROFILE_NAME.toString(),
+                this.provisioningProfileName);
 
         return properties;
     }
 
     protected void addProperty(Map<String, String> properties, String key, String value) {
+
         if (properties != null && key != null && value != null) {
             properties.put(key, value);
         }
     }
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-         this.properties = prepareProperties();
+
+        this.properties = prepareProperties();
+
+        try {
+            String projectName = Utils.buildProjectName(properties, mavenProject);
+            String schemeName = properties.get(Utils.PLUGIN_PROPERTIES.SCHEME.toString());
+            File projectDirectory = Utils.getWorkDirectory(properties, mavenProject, projectName);
+            this.currentXcodeVersion = Utils.getCurrentXcodeVersion(projectDirectory);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        if (properties.get(Utils.PLUGIN_PROPERTIES.XCODE_VERSION.toString()) != null && !properties.get(
+                Utils.PLUGIN_PROPERTIES.XCODE_VERSION.toString()).isEmpty()) {
+            try {
+                selectXcodeVersion(properties.get(Utils.PLUGIN_PROPERTIES.XCODE_VERSION.toString()),
+                        Utils.getWorkDirectory(properties, mavenProject, projectName));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        if (this.provisioningProfileName != null && !this.provisioningProfileName.equals("")) {
+            ProvisioningProfileHelper helper = new ProvisioningProfileHelper(this.provisioningProfileName, properties,
+                    mavenProject);
+            try {
+                ProvisioningProfileData data = helper.getData();
+                this.xcodeExportOptions.provisioningProfiles = new HashMap<>();
+                this.xcodeExportOptions.provisioningProfiles.put("bundleIdentifier", data.getUuid());
```

Comment:
hier muss statt dem String "bundleIdentifier" der wirkliche Bundle Identifier der App gesetzt werden. Den Wert dazu hast du hier im BaseMojo mit this.bundleIdentifier